openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -332,7 +334,10 @@ def get(self, request, usage_key_string, *args, **kwargs):
         """
         Return response to a GET request.
         """
-        usage_key = UsageKey.from_string(usage_key_string)
+        try:
+            usage_key = UsageKey.from_string(usage_key_string)
+        except InvalidKeyError:
+            raise NotFound("Invalid usage key: '{}'.".format(usage_key_string))
```

Comment:
I'm assuming there is existing handling for `NotFound`, but no such handling existed for `InvalidKeyError`?

Reply:
@stvstnfrd ``NotFound`` is a DRF specific exception which returns ``404`` response.